java-source/src/main/java/net/corda/examples/obligation/flows/SettleObligation.java

```diff
@@ -77,7 +77,7 @@ public SignedTransaction call() throws FlowException {
             final Party lenderIdentity = resolveIdentity(inputObligation.getLender());
 
             // Stage 3. This flow can only be initiated by the current recipient.
-            if (getOurIdentity() != borrowerIdentity) {
+            if (!borrowerIdentity.equals(getOurIdentity())) {
```

Comment:
This was comparing equality by reference which was returning false rather than comparing keys